src/rgw/rgw_kms.cc

```diff
@@ -1083,11 +1092,79 @@ static int get_actual_key_from_kmip(const DoutPrefixProvider *dpp,
     return -EINVAL;
   }
 }
+
 class KMSContext : public SSEContext {
-  CephContext *cct;
-public:
-  KMSContext(CephContext*_cct) : cct{_cct} {};
-  ~KMSContext() override {};
+  CephContext* cct;
+
+ public:
+  struct KMSCachedSecret {
+    size_t len;
+    key_serial_t serial;
+    KMSCachedSecret(const std::string& key, const std::string& secret)
+        : len(secret.size()) {
+      serial = ::add_key(
+          "user", key.c_str(), secret.c_str(), secret.size(),
+          KEY_SPEC_PROCESS_KEYRING);
+      ceph_assert(serial != -1);
```

Comment:
i think we'll probably want real error handling for these syscalls to avoid taking down the whole process

on error, we can always disable the cache as a fallback